perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/pwdmgr/EinfraPasswordManagerModule.java

```diff
@@ -67,6 +68,24 @@ public EinfraPasswordManagerModule() {
 
 	}
 
+	@Override
+	public String handleSponsorship(PerunSession sess, SponsoredUserData userData) {
+		if (StringUtils.isBlank(userData.getPassword())) {
+			throw new InternalErrorException("Einfra namespace requires a password.");
+		}
+		if (StringUtils.isBlank(userData.getNamespace())) {
```

Comment:
`reservePassword` itself performs checks on login and password based on namespace requirements, we shouldn't do it twice. Also you check namespace here but exception message mentions login.

Reply:
Good point. I have removed it.